rs/tests/driver/src/driver/ic_images.rs

```diff
@@ -0,0 +1,126 @@
+//! The following are helpers for tests that use ICOS images. Each artifact has the triplet (version, URL, hash).
+
+use crate::driver::test_env_api::read_dependency_from_env_to_string;
+use anyhow::{Context, Result};
+use ic_protobuf::registry::replica_version::v1::GuestLaunchMeasurements;
+use ic_types::ReplicaVersion;
+use url::Url;
+
+/// Pull the version of the initial GuestOS image from the environment.
+pub fn get_guestos_img_version() -> Result<ReplicaVersion> {
```

Comment:
Since this PR is just about moving code the following is better done in another PR but I would drop the `Result` from all of these functions and just panic. The first thing all callers do currently is an `.expect("...")` so better to just abstract that.